pkg/helm/controller.go

```diff
@@ -150,21 +164,46 @@ func (c *Controller) OnHelmRemove(key string, chart *helmv1.HelmChart) (*helmv1.
 	return newChart, c.apply.WithOwner(newChart).Apply(objectset.NewObjectSet())
 }
 
+func (c *Controller) OnConfChange(key string, conf *helmv1.HelmChartConfig) (*helmv1.HelmChartConfig, error) {
+	if chart, err := c.helmController.Cache().Get(conf.Namespace, conf.Name); err != nil {
+		if !errors.IsNotFound(err) {
+			return conf, err
+		}
+	} else if chart != nil {
+		if _, err := c.OnHelmChange(key, chart); err != nil && err != apply.ErrReplace {
+			logrus.Errorf("%s failed to trigger HelmChart update for %s: %s", Name, key, err)
+		}
+	}
+	return conf, nil
+}
+
+func (c *Controller) OnConfRemove(key string, conf *helmv1.HelmChartConfig) (*helmv1.HelmChartConfig, error) {
```

Comment:
In the past with wrangler this would add a finalizer which can prevent and or drastically slow down deletion of CR. Do we need OnRemove logic?

Reply:
If we decide to keep the OnRemove, check for and bail early if `conf == nil`. see  https://github.com/rancher/helm-controller/pull/59/files#r469447794

Reply:
We need to catch removal of the HelmChartConfig so that we can trigger an update of the HelmChart to remove the additional settings. I'll add a check for nil conf.

Reply:
I'm removing this, as per our Slack conversation.